app/Extensions/PaymentGateways/CryptoBot/CryptoBotExtension.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+namespace App\Extensions\PaymentGateways\CryptoBot;
+
+use App\Classes\AbstractExtension;
+use App\Enums\PaymentStatus;
+use App\Events\PaymentEvent;
+use App\Events\UserUpdateCreditsEvent;
+use App\Models\PartnerDiscount;
+use App\Models\Payment;
+use App\Models\ShopProduct;
+use App\Models\User;
+use App\Models\Coupon;
+use App\Traits\Coupon as CouponTrait;
+use App\Events\CouponUsedEvent;
+use Exception;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Redirect;
+use Illuminate\Support\Facades\Log;
+use Illuminate\Support\Facades\Http;
+use App\Notifications\ConfirmPaymentNotification;
+
+/**
+ * Summary of PayPalExtension
+ */
+class CryptoBotExtension extends AbstractExtension
+{
+    use CouponTrait;
+
+    public static function getConfig(): array
+    {
+        return [
+            "name" => "CryptoBot",
+            "RoutesIgnoreCsrf" => [
+                "payment/CryptoBotWebhook"
+            ],
+        ];
+    }
+
+    public static function getRedirectUrl(Payment $payment, ShopProduct $shopProduct, string $totalPriceString): string
+    {
+        $url = 'https://pay.crypt.bot/api/createInvoice';
+        $settings = new CryptoBotSettings();
+        try {
+            $response = Http::withHeaders([
+                'Accept' => 'application/json',
+                'Content-Type' => 'application/json',
+                'Crypto-Pay-API-Token' => $settings->api_key
+            ])->post($url, [
+                'amount' => $totalPriceString,
+                'payload' => strval($payment->id),
+                'description' => "Заказ #{$payment->id} - " . $shopProduct->name,
+                'currency_type' => 'fiat',
+                'fiat' => $shopProduct->currency_code,
+                'hidden_message' => $settings->hidden_message,
+                'paid_btn_name' => 'callback',
+                'paid_btn_url' => route('payment.CryptoBotSuccess').'?payment=YGqvOf4I'
```

Comment:
I'm not quite sure if it's correct, can you double-check this part of the code? To be more precise, this -> `?payment=YGqvOf4I`